02_bubble/lexual/sort_test.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+// table driven tests
+var tests = []struct {
+	input    []int
+	expected []int
+}{
+	{
+		[]int{3, 2, 1, 5},
+		[]int{1, 2, 3, 5},
+	},
+	{
+		[]int{},
+		[]int{},
+	},
+	{
+		[]int{1},
+		[]int{1},
+	},
+	{
+		[]int{1, 2},
+		[]int{1, 2},
+	},
+}
+
+func TestBubbleSort(t *testing.T) {
+	for _, test := range tests {
+		testSort(t, bubble(test.input), test.expected)
+	}
+}
+
+func TestInsertionSort(t *testing.T) {
+	for _, test := range tests {
+		testSort(t, insertion(test.input), test.expected)
+	}
+}
+
+func testSort(t *testing.T, actual []int, expected []int) {
```

Comment:
can use `require.Equal(t, expected, calculated)` instead, although this wont give you the custom error output